### PR TITLE
Containerized esi-leap

### PR DESCRIPTION
**Pull request by rd4398**
This is a basic skeleton for the container file which can be used for containerized esi-leap. I referred to following links:

https://docs.docker.com/engine/reference/builder/
https://www.pybootcamp.com/blog/how-to-containerize-python-application/
https://people.redhat.com/~jupittma/cheatsheets/podman.html
https://www.pybootcamp.com/blog/how-to-write-dockerfile-python-apps/
https://snyk.io/blog/best-practices-containerizing-python-docker/

I need your inputs for further editing this specifically keeping esi-leap in mind

### CODE REVIEW

**Comment by rd4398** [comment]
Also, not sure why PR checks are failing...
I did not change any code, just added a new file named 'containerfile'!

File: Containerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+MAINTAINER <tzumainn@redhat.com>
+
+# The working directory for esi-leap inside the container
+
+WORKDIR /esi-leap
+
+# Copy the files of esi leap inside the container
+
+COPY . .
```
Comment (DanNiESh) [inline Containerfile:21]: Can you add the files that need to be copied to the container in this repo? I think esi-leap.conf should be included and saved under /etc/esi-leap.

File: Containerfile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+RUN pip install esi-leap
+
+# Something for esi-leap-api and esi-leap-manager
+# Not sure how to link these two here / bring inside the container
+
+# Start processes for esi-leap-api and esi-leap-manager
+
+# What is the entry point / executable for esi-leap?
+
+CMD ["python", ./executable]
```
Comment (DanNiESh) [inline Containerfile:34]: Maybe use the commands here? https://github.com/CCI-MOC/esi-leap#run-the-services

File: Containerfile [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+LABEL description="ESI-LEAP container file"
+
+# Maintainer details
+
+MAINTAINER <tzumainn@redhat.com>
+
+# The working directory for esi-leap inside the container
+
+WORKDIR /esi-leap
+
```
Comment (DanNiESh) [inline Containerfile:18]: You could add an "EXPOSE <port>" line in Dockerfile to expose esi-leap-api port

**Comment by rd4398** [comment]
Okay, will do
Can you tell me on which port esi-leap-api runs?

**Comment by DanNiESh** [comment]
> Okay, will do Can you tell me on which port esi-leap-api runs?

It's 7777 by default. https://github.com/CCI-MOC/esi-leap/blob/master/esi_leap/conf/api.py

**Comment by rd4398** [comment]
Ah, okay!

**Comment by rd4398** [comment]
The container build is failing for the command : "sudo esi-leap-dbsync create_schema" with error as "sudo command not found".... Any work around for this?
I guess we need this command to run inside the container....

**Comment by DanNiESh** [comment]
> sudo command not found

Can you try to add "USER root" in dockerfile before "RUN sudo esi-leap-dbsync create_schema". I think you need to be root user to run esi-leap-xxx.

> The container build is failing for the command : "sudo esi-leap-dbsync create_schema" with error as "sudo command not found".... Any work around for this? I guess we need this command to run inside the container....

**Comment by rd4398** [comment]
Okay, will try!

**Comment by rd4398** [comment]
I am using '$COPY . .' in container file which copies entire esi-leap folder in the container, which is making me believe that esi-leap.conf will also be getting copied in the container. I am not able to figure out why I am still getting 'sudo command not found error' for "esi-leap-dbsync create schema". This needs to be figured out....

**Pull request closed without merging**